src/main/java/org/openjdk/jextract/impl/ClassSourceBuilder.java

```diff
@@ -50,160 +47,104 @@ enum Kind {
         }
     }
 
-    final Kind kind;
-    final ClassDesc desc;
-    protected final JavaSourceBuilder enclosing;
+    private final SourceFileBuilder sb;
+    private final List<String> enclosingClassNames;
+    private final String modifiers;
+    private final Kind kind;
+    private final String className;
+    private final String superName;
 
-    // code buffer
-    private StringBuilder sb = new StringBuilder();
-    // current line alignment (number of 4-spaces)
-    private int align;
-
-    ClassSourceBuilder(JavaSourceBuilder enclosing, Kind kind, String name) {
-        this.enclosing = enclosing;
-        this.align = (enclosing instanceof ClassSourceBuilder classSourceBuilder)
-                ? classSourceBuilder.align : 0;
+    ClassSourceBuilder(SourceFileBuilder builder, String modifiers, Kind kind, String className, String superName, List<String> enclosingClassNames) {
+        this.sb = builder;
+        this.modifiers = modifiers;
         this.kind = kind;
-        this.desc = ClassDesc.of(enclosing.packageName(), name);
+        this.className = className;
+        this.superName = superName;
+        this.enclosingClassNames = enclosingClassNames;
     }
 
-    boolean isNested() {
-        return enclosing instanceof ClassSourceBuilder;
+    final String className() {
+        // for a (nested) class 'com.foo.package.A.B.C' this will return 'C'
+        return className;
     }
 
-    String className() {
-        return desc.displayName();
```

Comment:
Note that `desc.displayName()` is inaccurate for nested classes, as it will return e.g. `A$B` for a nested `A.B` class. This is then impossible to distinguish from class names that actually have a `$` in the name. I felt that it was better to just use a `String` to represent the simple class name instead.